From: Vivien Didelot <vivien.didelot@savoirfairelinux.com>
Subject: net: dsa: legacy: don't unmask port bitmaps
Patch-mainline: v4.15-rc1
Git-commit: 8e8ef50bb4246d6fee9971994a6d846d9bc7ff4c
References: git-fixes

The legacy code does not unmask the cpu_port_mask and dsa_port_mask as
stated. But this is done on the error path and those masks won't be used
after that. So instead of fixing the bit operation, simply remove it.

Fixes: 83c0afaec7b7 ("net: dsa: Add new binding implementation")
Signed-off-by: Vivien Didelot <vivien.didelot@savoirfairelinux.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Denis Kirjanov <dkirjanov@suse.com>
---
 net/dsa/legacy.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/net/dsa/legacy.c b/net/dsa/legacy.c
index 7281098df04e..957510550b1e 100644
--- a/net/dsa/legacy.c
+++ b/net/dsa/legacy.c
@@ -277,10 +277,6 @@ static void dsa_switch_destroy(struct dsa_switch *ds)
 		if (!(dsa_is_cpu_port(ds, port) || dsa_is_dsa_port(ds, port)))
 			continue;
 		dsa_cpu_dsa_destroy(&ds->ports[port]);
-
-		/* Clearing a bit which is not set does no harm */
-		ds->cpu_port_mask |= ~(1 << port);
-		ds->dsa_port_mask |= ~(1 << port);
 	}
 
 	if (ds->slave_mii_bus && ds->ops->phy_read)
-- 
2.12.3

